specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2021-07-22-preview/PrometheusRuleGroups.json

```diff
@@ -0,0 +1,601 @@
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "Last time the rule was updated in ISO8601 format."
+        }
+      },
+      "description": "An alert rule."
+    },
+    "PrometheusRuleGroupResource": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PrometheusRuleGroupProperties",
+          "description": "The Prometheus rule group properties of the resource."
+        },
+        "systemData": {
+          "description": "Action rule system data",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "description": "The Prometheus rule group resource."
+    },
+    "PrometheusRuleGroupResourcePatch": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PrometheusRuleGroupProperties",
+          "description": "The alert rule properties of the resource."
+        }
+      },
+      "description": "The Prometheus rule group resource for patch operations."
+    },
+    "PrometheusRuleGroupResourceCollection": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrometheusRuleGroupResource"
+          },
+          "description": "the values for the alert rule resources."
+        }
+      },
+      "description": "Represents a collection of alert rule resources."
+    },
+    "PrometheusRule": {
+      "type": "object",
+      "properties": {
+        "record": {
+          "type": "string",
+          "description": "the name of the recording rule."
+        },
+        "alert": {
+          "type": "string",
+          "description": "the name of the alert rule."
+        },
+        "expression": {
+          "type": "string",
+          "description": "the expression to run for the rule."
+        },
+        "severity": {
+          "type": "integer",
+          "format": "int32",
+          "description": "the expression to run for the rule."
+        },
+        "for": {
+          "type": "string",
+          "description": "the amount of time alert must be active before firing."
+        },
+        "labels": {
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

ARM recommendation is to avoid this pattern wherever possible, since it doesn't allow reasoning about the payload (so things like Azure Policy and ARG can't use them). Same for annotations. If these properties are always only user-provided, then it won't block you, but it's still worth considering other options wherever possible.

Reply:
Agreed - but again this is due to alignment with Prometheus: https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/